registry/base_registry.go

```diff
@@ -178,7 +178,10 @@ func (r *BaseRegistry) RestartCallBack() bool {
 		}
 		logger.Infof("success to re-register service :%v", confIf.Key())
 	}
-	r.facadeBasedRegistry.InitListeners()
+
+	if flag {
+		r.facadeBasedRegistry.InitListeners()
+	}
```

Comment:
This flag will make the listener's pathMap of registry can not be cleaned every time.

Reply:
> This flag will make the listener's pathMap of registry can not be cleaned every time.

good job.

Reply:
> This flag will make the listener's pathMap of registry can not be cleaned every time.

this is not  the main problem